prediction_market_agent/agents/crewai_subsequential_agent/crewai_agent_subquestions.py

```diff
@@ -0,0 +1,140 @@
+            tools=[search_tool],
+        )
+
+        self.analyst = Agent(
+        role='Data Analyst',
+        goal='Analyze research findings',
+        backstory='A meticulous analyst with a knack for uncovering patterns'
+        )
+
+        self.writer = Agent(
+        role='Writer',
+        goal='Draft the final report',
+        backstory='A skilled writer with a talent for crafting compelling narratives'
+        )
+
+        self.predictor = Agent(
+                    role="Professional Gambler",
+                    goal="Predict, based on some research you are presented with, whether or not a given event will occur",
+                    backstory="You are a professional gambler who is adept at predicting and betting on the outcomes of future events.",
+                    verbose=True,
+                    allow_delegation=False,
+                    tools=[]
+                )
+
+    def split_research_into_outcomes(self, question: str) -> Outcomes:
+        create_outcomes_task = Task(
+          description=(CREATE_OUTCOMES_FROM_SCENARIO_PROMPT),
+          expected_output=CREATE_OUTCOMES_FROM_SCENARIO_OUTPUT,
+          tools = [],
+          output_json=Outcomes,
+          agent=self.researcher,
+        )
+
+        report_crew = Crew(
+            agents=[self.researcher],
+            tasks=[create_outcomes_task],
+        )
+        result = report_crew.kickoff(inputs={'scenario': question})
+        return result
+
+    def generate_prediction_for_one_outcome(self, sentence: str) -> ProbabilityOutput:
+
+        task_research_one_outcome = Task(
+                    description=(RESEARCH_OUTCOME_PROMPT),
+                    tools=[search_tool],
+                    agent=self.researcher,
+                    expected_output=(RESEARCH_OUTCOME_OUTPUT),
+                )
+        task_create_probability_for_one_outcome = Task(
+                    description=(PROBABILITY_FOR_ONE_OUTCOME_PROMPT),
+                expected_output=PROBABILITY_CLASS_OUTPUT,
+                    agent=self.predictor,
+                output_json=ProbabilityOutput,
+            context=[task_research_one_outcome]
+                )
+        crew = Crew(
+                    agents=[self.researcher, self.predictor],
+                    tasks=[task_research_one_outcome, task_create_probability_for_one_outcome],
+                    verbose=2,
+                process=Process.sequential            
+                )
+        
+        result = crew.kickoff(inputs={'sentence': sentence})
+        return result
+
+    def generate_final_decision(self, outcomes_with_probabilities) -> ProbabilityOutput:
+
+        task_final_decision = Task(
+                    description=(FINAL_DECISION_PROMPT),
+                    tools=[],
+                    agent=self.predictor,
+                    expected_output=(PROBABILITY_CLASS_OUTPUT),
+                output_json=ProbabilityOutput,
+                )
+        
+        crew = Crew(
```

Comment:
What does the sequence of LLM calls look like when you have a 1-agent crew?

Reply:
They are basically processed sequentially - as you correctly pointed out below, I am "manually" looping after one of the LLM calls and doing N calls (1 per outcome) to the researchar and gambler agents to come up with a prediction for each outcome.
This should be parallelized - I have an idea on how to do so, implementing it now.

Reply:
Implement a parallelized version usinc `async_execution=True`

Reply:
Ah cool, I wasn't thinking of parallelization, more just what the pattern of conversation looks like in a 1-person crew. i.e. the predictor is given a task, then it outputs its first batch of tokens. What happens next?